### DISCUSSION TITLE
Is it possible to use this with my custom python code (not HA type stuff)?

### DISCUSSION THREAD

**Discussion by chrisaccount** [krahabb/meross_lan#523] [Q&A] [answered]
I have been looking at the fantastic code/documentation on the meross_lan github and wiki.  So first of all thanks very much for that.
i have custom python stuff i'm playing with and just bought the MSG 100 garage door opener.  It is working fine with official eWiseHome stuff.  I'm seeking the easiest path to get my python stuff to control (and monitor) the garage door.  Wondering if meross_lan code from git hub could help me with this (or is it specific to HA and would need lots of changes)?  Any thoughts appreciated.

**Comment by krahabb** [comment, marked as answer]
Hello @chrisaccount,
Thank you for your appreciation.
Not sure if it would be easy since it was designed from the beginning as an HA component but I've tried to isolate 'library' code in the `merossclient` folder which does not depend at all on homeassistant code.
Digging into that module it should be easy to setup an HTTP client (even an MQTT one if that matters).

I remember I've made some experiments in the past due to a specific request but I cannot find the thread where this was discussed.
In the end, I remember it was just a matter of properly 'loading' the folder as a python module into your code and use it.

**Reply by chrisaccount** [reply]
Thanks so much for your thoughts on this.  I will give it a go and post results.
